docs/arch.md

```diff
@@ -87,23 +87,18 @@ There are these main tasks scheduled:
 ### Scheduling of Gatherer
 Gatherer is using this logic to start information gathering from the cluster, and it is handled in [periodic.go](pkg/controller/periodic/periodic.go).
 
-The periodic is using a producer/consumer queue. It is periodically adding the gatherer to queue. The queue has a limit for maximally one gatherer of the name in queue and because we only have gatherer "config" it is only this one in queue.
-The adding to queue is run in periodic.periodicTrigger, which is started from periodic.Run function. 
-Function periodicTrigger tries to add a new one (blocking if the queue is full) every interval/2.
+So far we have only 1 Gatherer(called `clusterconfig`), it has several gather-functions each collecting different data from the cluster.
+The workflow of the gather-functions is managed by the Gatherer.
+Only one Gatherer runs at one time, this is because we only have 1 Gatherer at the moment. (ie.: we can and concurrency here when its needed)
+When IO starts there is an initial delay before the first `Gather` happens, after that a `Gather` is initiated every interval, this is done by `periodicTrigger`.
+`periodic.Run` handles the initial delay and starts the `periodicTrigger` like `go wait.Until(func() { c.periodicTrigger(stopCh) }, time.Second, stopCh)`.
 
-Code: In periodic.Run it is calling periodicTrigger if it has a chance
-```
-go wait.Until(func() { c.periodicTrigger(stopCh) }, time.Second, stopCh)
-```
-The periodicTrigger is special in a way that it tries to balance times when it inserts to the queue.
-`wait.Jitter(i, m)` returns `time.Duration` after `time i + random(i*m)`.
-And even if it is time to add to queue it balances inserts with AddAfter between `i/4 to i/4 + random(i/4 * 2)` (sometimes between 1/4 and 3/4).
-
-The consumers from the queue are 4 workers started as goroutines from periodic.Run function.
-Because we only have one gatherer in queue, I think only one worker will effectively run task from queue. The actual start of gatherer is call to c.sync from periodic.processNextWorkItem. The c.sync is also defined in periodic and calls Gather + immediately stores returned data to disk as one file with timestamp.
+`Gather` uses `ExponentialBackoff` to retry (amount specified in: `status.GatherFailuresCountThreshold`) if a Gatherer returns any errors, these errors are mostly caused when a collected resource is not yet ready therefore it can't be right now collected so we should retry later.
+Its important that all retries needs to finish before the next gather period starts, so that we don't have potential conflicts, the Backoff is calibrated to take this into account.
```

Comment:
`It's important that all retries need to finish..` or maybe `It's important that all retries have to finish..` ?

Reply:
What about: `It's important that all retries finish..`